kernel_tuner/backends/hip.py

```diff
@@ -85,6 +86,8 @@ def __init__(self, device=0, iterations=7, compiler_options=None, observers=None
         for obs in self.observers:
             obs.register_device(self)
 
+    def allocate_ndarray(self, array):
+        return hip.hipMalloc(array.nbytes)
```

Comment:
Don't you need to store the allocated memory?
```suggestion
    def allocate_ndarray(self, array):
        alloc = hip.hipMalloc(array.nbytes)
        self.allocations.append(alloc)
        return alloc
```